tools/topology/topology2/cavs/cavs-nocodec.conf

```diff
@@ -116,13 +118,36 @@ Object.Dai {
 #
 # Pipeline definitions
 #
+# PCM0 ---> gain ----> Mixin ----> Mixout ----> gain ----> SSP0
+# PCM1 ---> gain ----> Mixin ----> Mixout ----> gain ----> SSP1
+# PCM2 ---> gain ----> Mixin ----> Mixout ----> gain ----> SSP2
+#
+# SSP0 ---> gain ----> PCM0
+# SSP1 ---> gain ----> PCM1
+# SSP2 ---> gain ----> PCM2
 
 # Pipeline ID:1 PCM ID: 0
 Object.Pipeline {
 	# playback pipelines
-	passthrough-be.1 {
-		index	2
-		direction	playback
+	host-copier-gain-mixin-playback.1 {
+		index	1
+
+		Object.Widget.pipeline.1 {
+			stream_name 'NoCodec-0'
+		}
+		Object.Widget.copier.1 {
+			stream_name 'SSP0 Playback'
+		}
+		Object.Widget.gain.1 {
+			Object.Control.mixer.1 {
+				name 'Playback Volume 0'
```

Comment:
why "Playback Volume 0"? Shouldnt it be "Playback Volume 1" to match the pipeline ID and similarly for the other gain controls as well below

Reply:
sure, good idea. BTW we can't use pipeline 0 and need to start with pipeline 1 ?

Reply:
Yes, its a tplg2 requirement. All nodes should preferably start with ID 1 instead of 0.